x-pack/plugins/global_search_providers/server/providers/saved_objects/get_searchable_types.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ISavedObjectTypeRegistry, SavedObjectsType } from 'src/core/server';
+
+export const getSearchableTypes = (typeRegistry: ISavedObjectTypeRegistry, types?: string[]) => {
+  const typeFilter = types
+    ? (type: SavedObjectsType) => {
+        if (type.management?.displayName && isTypeMatching(types, type.management.displayName)) {
+          return true;
+        }
+        return isTypeMatching(types, type.name);
+      }
+    : () => true;
+
+  return typeRegistry
+    .getVisibleTypes()
+    .filter(typeFilter)
+    .filter((type) => type.management?.defaultSearchField && type.management?.getInAppUrl);
+};
+
+const isTypeMatching = (list: string[], item: string) =>
+  list.find((e) => toCompareFormat(e) === toCompareFormat(item)) !== undefined;
+
+const toCompareFormat = (str: string) => str.trim().toLowerCase().replace(/\s/g, '-');
```

Comment:
I'm worried that we need to normalize the input value on every call and potentially in several places. Can we move at least `trim` closer to the input source?